flink-runtime/src/main/java/org/apache/flink/runtime/filecache/FileCache.java

```diff
@@ -179,27 +179,40 @@ public Path call()  {
 		
 		private String name;
 		private JobID jobID;
-		private int oldCount;
+		private String filePath;
 
 		public DeleteProcess(String name, DistributedCacheEntry e, JobID jobID, int c) {
 			this.name = name;
 			this.jobID = jobID;
-			this.oldCount = c;
+			this.filePath = e.filePath;
 		}
+
 		@Override
 		public void run() {
-			synchronized (count) {
-				if (count.get(new ImmutablePair<JobID, String>(jobID, name)) != oldCount) {
+			Pair<JobID, String> key = new ImmutablePair<JobID, String>(jobID, name);
+			synchronized (lock) {
+				if (!count.containsKey(key)) {
 					return;
```

Comment:
Could we invert the if condition? Imho something like ```if(count.containsKey(key)){ .... }``` gives a cleaner control flow without too many return statements.